src/main/java/com/wimbli/WorldBorder/BorderData.java

```diff
@@ -291,26 +291,102 @@ public Location correctedPosition(Location loc)
 	}
 
 	//these material IDs are acceptable for places to teleport player; breathable blocks and water
-	public static final LinkedHashSet<Integer> safeOpenBlocks = new LinkedHashSet<Integer>(Arrays.asList(
-		 new Integer[] {0, 6, 8, 9, 27, 28, 30, 31, 32, 37, 38, 39, 40, 50, 55, 59, 63, 64, 65, 66, 68, 69, 70, 71, 72, 75, 76, 77, 78, 83, 90, 93, 94, 96, 104, 105, 106, 115, 131, 132, 141, 142, 149, 150, 157, 171}
-	));
+	public static final EnumSet<Material> safeOpenBlocks = EnumSet.noneOf(Material.class);
+	static
+	{
+		safeOpenBlocks.add(Material.AIR);
+		safeOpenBlocks.add(Material.CAVE_AIR);
+		safeOpenBlocks.add(Material.OAK_SAPLING);
+		safeOpenBlocks.add(Material.SPRUCE_SAPLING);
+		safeOpenBlocks.add(Material.BIRCH_SAPLING);
+		safeOpenBlocks.add(Material.JUNGLE_SAPLING);
+		safeOpenBlocks.add(Material.ACACIA_SAPLING);
+		safeOpenBlocks.add(Material.DARK_OAK_SAPLING);
+		safeOpenBlocks.add(Material.WATER);
+		safeOpenBlocks.add(Material.RAIL);
+		safeOpenBlocks.add(Material.POWERED_RAIL);
+		safeOpenBlocks.add(Material.DETECTOR_RAIL);
+		safeOpenBlocks.add(Material.ACTIVATOR_RAIL);
+		safeOpenBlocks.add(Material.COBWEB);
+		safeOpenBlocks.add(Material.GRASS);
+		safeOpenBlocks.add(Material.FERN);
+		safeOpenBlocks.add(Material.DEAD_BUSH);
+		safeOpenBlocks.add(Material.DANDELION);
+		safeOpenBlocks.add(Material.POPPY);
+		safeOpenBlocks.add(Material.BLUE_ORCHID);
+		safeOpenBlocks.add(Material.ALLIUM);
+		safeOpenBlocks.add(Material.AZURE_BLUET);
+		safeOpenBlocks.add(Material.RED_TULIP);
+		safeOpenBlocks.add(Material.ORANGE_TULIP);
+		safeOpenBlocks.add(Material.WHITE_TULIP);
+		safeOpenBlocks.add(Material.PINK_TULIP);
+		safeOpenBlocks.add(Material.OXEYE_DAISY);
+		safeOpenBlocks.add(Material.BROWN_MUSHROOM);
+		safeOpenBlocks.add(Material.RED_MUSHROOM);
+		safeOpenBlocks.add(Material.TORCH);
+		safeOpenBlocks.add(Material.WALL_TORCH);
+		safeOpenBlocks.add(Material.REDSTONE_WIRE);
+		safeOpenBlocks.add(Material.WHEAT);
+		safeOpenBlocks.add(Material.SIGN);
+		safeOpenBlocks.add(Material.WALL_SIGN);
+		safeOpenBlocks.add(Material.LADDER);
+		safeOpenBlocks.add(Material.LEVER);
+		safeOpenBlocks.add(Material.LIGHT_WEIGHTED_PRESSURE_PLATE);
+		safeOpenBlocks.add(Material.HEAVY_WEIGHTED_PRESSURE_PLATE);
+		safeOpenBlocks.add(Material.STONE_PRESSURE_PLATE);
+		safeOpenBlocks.add(Material.OAK_PRESSURE_PLATE);
+		safeOpenBlocks.add(Material.SPRUCE_PRESSURE_PLATE);
+		safeOpenBlocks.add(Material.BIRCH_PRESSURE_PLATE);
+		safeOpenBlocks.add(Material.JUNGLE_PRESSURE_PLATE);
+		safeOpenBlocks.add(Material.ACACIA_PRESSURE_PLATE);
+		safeOpenBlocks.add(Material.DARK_OAK_PRESSURE_PLATE);
+		safeOpenBlocks.add(Material.REDSTONE_TORCH);
+		safeOpenBlocks.add(Material.REDSTONE_WALL_TORCH);
+		safeOpenBlocks.add(Material.STONE_BUTTON);
```

Comment:
Maybe add the wood buttons, too?

Reply:
Good idea, I had just transformed the old list, but now I added some new blocks that were not in that list.